doc/development/adding_operators.rst

```diff
@@ -0,0 +1,360 @@
+     ((1.0, 2.0), [PauliX(wires=[0]), PauliZ(wires=[0])])
+
+   * Representation via the **eigenvalue decomposition** specified by eigenvalues (for the diagonal matrix, :meth:`.Operator.eigvals`)
+     and diagonalizing gates (for the unitaries :meth:`.Operator.diagonalizing_gates`):
+
+     >>> op = qml.PauliX(0)
+     >>> op.diagonalizing_gates()
+     [Hadamard(wires=[0])]
+     >>> op.eigvals()
+     [ 1 -1]
+
+   * Representation as a **matrix** (:meth:`.Operator.matrix`), as specified by a global wire order that tells us where the
+     wires are found on a register:
+
+     >>> op = qml.PauliRot(0.2, "X", wires=["b"])
+     >>> op.matrix(wire_order=["a", "b"])
+     [[9.95e-01-2.26e-18j 2.72e-17-9.98e-02j, 0+0j, 0+0j]
+      [2.72e-17-9.98e-02j 9.95e-01-2.26e-18j, 0+0j, 0+0j]
+      [0+0j, 0+0j, 9.95e-01-2.26e-18j 2.72e-17-9.98e-02j]
+      [0+0j, 0+0j, 2.72e-17-9.98e-02j 9.95e-01-2.26e-18j]]
+
+   * Representation as a **sparse matrix** (:meth:`.Operator.sparse_matrix`):
+
+     >>> from scipy.sparse.coo import coo_matrix
+     >>> row = np.array([0, 1])
+     >>> col = np.array([1, 0])
+     >>> data = np.array([1, -1])
+     >>> mat = coo_matrix((data, (row, col)), shape=(4, 4))
+     >>> op = qml.SparseHamiltonian(mat, wires=["a"])
+     >>> op.sparse_matrix(wire_order=["a"])
+     (0, 1)   1
+     (1, 0) - 1
+
+New operators can be created by applying arithmetic functions to operators, such as addition, scalar multiplication,
+multiplication, taking the adjoint, or controlling an operator. At the moment, such arithmetic is only implemented for
+specific subclasses.
+
+* Operators inheriting from :class:`~.Observable` support addition and scalar multiplication:
+
+  >>> op = qml.PauliX(0) + 0.1 * qml.PauliZ(0)
+  >>> op.name
+  Hamiltonian
+  >>> op
+    (0.1) [Z0]
+  + (1.0) [X0]
+
+* Operators may define a hermitian conjugate:
+
+  >>> qml.RX(1., wires=0).adjoint()
+  RX(-1.0, wires=[0])
+
+Creating custom operators
+#########################
+
+A custom operator can be created by inheriting from :class:`~.Operator` or one of its subclasses.
+
+The following is an example for a custom gate that possibly flips a qubit and then rotates another qubit.
+The custom operator defines a decomposition, which the devices can use (since it is unlikely that a device
+knows a native implementation for ``FlipAndRotate``). It also defines an adjoint operator.
+
+.. code-block:: python
+
+    import pennylane as qml
+
+
+    class FlipAndRotate(qml.operation.Operation):
+        """One-sentence description of the operator.
+
+        Add more explanation. How is the operator defined, what are typical usage contexts?
+        What is the meaning of the different inputs? What options does a user have?
+
+        Args:
+            Inputs are described here
+
+        **Example**
+
+        Various code examples that explain how the operator is employed in practice.
+        It is recommended to add the examples as clearly marked tests to ensure their
```

Comment:
Does this imply we can automatically run tests from docstring examples? :)

Reply:
In the future, maybe! We should be able to use [`doctest`](https://docs.python.org/3/library/doctest.html).

Reply:
Ah, then I'm not quite sure what this sentence instructs me to do. Should the docstring examples also show up 1:1 as tests? What does clearly marked mean?

Reply:
I must admit I don't fully follow it either.

Reply:
Oh I added this on a whim...I stumbled over some test that was clearly marked as testing code used in a docstring and thought this is such a great idea. But it is not current practice. Shall I delete this?

Reply:
I'm in favour of removing the description from here, and instead linking to https://pennylane.readthedocs.io/en/stable/development/guide/documentation.html#functions-and-methods, just so that the 'source of truth' is only in one place